src/components/chat/chat_channels.jsx

```diff
@@ -276,6 +270,10 @@ const ChatChannels = React.createClass({
       channels: ChatChannelsStore.getChannels(),
       subscribedChannels: ChatChannelsStore.getSubscribedChannels()
     });
+    // Fetch the initial list of messages for all subscribed rooms.
+    ChatChannelsStore.getSubscribedChannels().map((channel) => {
```

Comment:
Also, if we decide that this is safe, we should move it to the `setState` callback and use `this.state.subscribedChannels`, I think.